### PR TITLE
Make the table remove/add the row/column-header-related css classes in updateSettings.

### PR DESCRIPTION
**Pull request by jansiegel**
### Context
The problem in #5735 was caused by `updateSettings` not updating the `rootElement` css classes with the current `rowHeaders`/`colHeaders` state. This PR fixed that.

### How has this been tested?
Added 2 test cases for adding/removing the css classes.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #5735 

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ://github.com/handsontable/handsontable/pull/7162)
 - Fixed an issue where the column headers were cut off after hiding and then showing columns using the Hidden Columns
  plugin. [#6395](https://github.com/handsontable/handsontable/issues/6395)
+- Fixed an issue where the rows were missing their left border after disabling the row headers using `updateSettings
+`, while there were `fixedColumnsLeft` defined. [#5735](https://github.com/handsontable/handsontable/issues/5735)
```
Comment (wszymanski) [inline CHANGELOG.md:20]: Please change to a single line.

**Review by wszymanski — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (jansiegel) [reply CHANGELOG.md:20]: Is this an official rule? @wojciechczerniak

Comment (wojciechczerniak) [reply CHANGELOG.md]: No, but sounds good to establish one. I've noticed that line breaks broke the links and formatting (code block had additional space). And might be formatted differently depending on container max width (when we move changelog to release notes on github or to a blog post), and on mobile devices when the github is not displayed at the default width.

Comment (jansiegel) [reply CHANGELOG.md]: > And might be formatted differently depending on container max width

I don't think a single line break creates a visual line break in markdown. But it's true, the auto hard break in my IDE broke the url to one of the entries, so maybe let's stick to not breaking lines at all.

**Review by wszymanski — DISMISSED** [review]

**Review by wszymanski — APPROVED** [review]

**Pull request merged**